--- Makefile	2015-01-05 09:31:33.000000000 +0100
+++ Makefile.emscripten	2015-01-05 09:31:38.000000000 +0100
@@ -97,6 +97,7 @@
 	context.cpp \
 	contextualize.cpp \
 	copy_c_str.cpp \
+	emscripten_wrapper.cpp \
 	error_handling.cpp \
 	eval.cpp \
 	expand.cpp \
@@ -167,6 +168,9 @@
 static: lib/libsass.a
 shared: $(LIBRARIES)
 
+js: static
+	emcc -O2 lib/libsass.a -o lib/libsass.js -s EXPORTED_FUNCTIONS="['_sass_compile_emscripten']" -s DISABLE_EXCEPTION_CATCHING=0 --memory-init-file 0
+
 lib/libsass.a: $(COBJECTS) $(OBJECTS)
 	$(MKDIR) lib
 	$(AR) rcvs $@ $(COBJECTS) $(OBJECTS)
@@ -217,7 +221,7 @@
 	$(RUBY_BIN) $(SASS_SPEC_PATH)/sass-spec.rb -c $(SASSC_BIN) $(LOG_FLAGS) $(SASS_SPEC_PATH)/spec/issues
 
 clean:
-	$(RM) $(RCOBJECTS) $(COBJECTS) $(OBJECTS) $(LIBRARIES) lib/*.a lib/*.so lib/*.dll lib/*.la
+	$(RM) $(RCOBJECTS) $(COBJECTS) $(OBJECTS) $(LIBRARIES) lib/*.a lib/*.so lib/*.dll lib/*.la lib/*.js
 
 
 .PHONY: all debug debug-static debug-shared static shared install install-static install-shared sassc clean
